matchingengine/OrderBook_test.go

```diff
@@ -70,23 +116,366 @@ func TestLoadOldAsk(t *testing.T) {
+
+	config := utils.GetConfiguration()
+	utils.Init(config)
+
+	mockControl, ob, _, mockBidQueue, _, _, mockDepth, stockID, stockQuantity, stockPrice := helperForOrderBookTests(t)
+	defer mockControl.Finish()
+
+	limitBid := makeBid(1, stockID, models.Limit, stockQuantity, stockPrice, "")
+
+	mockBidQueue.EXPECT().Push(limitBid).Times(1)
+	mockDepth.EXPECT().AddOrder(false, false, stockPrice, stockQuantity).Times(1)
+
+	ob.LoadOldBid(limitBid)
+
+}
+
+func TestLoadOldBidStopLoss(t *testing.T) {
+
+	config := utils.GetConfiguration()
+	utils.Init(config)
+
+	mockControl, ob, _, _, _, mockBidStoplossQueue, _, stockID, stockQuantity, stockPrice := helperForOrderBookTests(t)
+	defer mockControl.Finish()
+
+	stopLossBid := makeBid(1, stockID, models.StopLoss, stockQuantity, stockPrice, "")
+
+	mockBidStoplossQueue.EXPECT().Push(stopLossBid).Times(1)
+
+	ob.LoadOldBid(stopLossBid)
+
+}
+
+func TestLoadOldTransactions(t *testing.T) {
+
+	config := utils.GetConfiguration()
+	utils.Init(config)
+
+	mockControl, ob, _, _, _, _, mockDepth, stockID, _, _ := helperForOrderBookTests(t)
 	defer mockControl.Finish()
 
+	t1 := &models.Transaction{
+		Id:            1,
+		UserId:        10,
+		StockId:       stockID,
+		Type:          1,
+		StockQuantity: 20,
+		Price:         50,
+		Total:         1000,
+		CreatedAt:     "random1",
+	}
+
+	t2 := &models.Transaction{
+		Id:            2,
+		UserId:        10,
+		StockId:       stockID,
+		Type:          2,
+		StockQuantity: 10,
+		Price:         20,
+		Total:         200,
+		CreatedAt:     "random2",
+	}
+
+	transactions := []*models.Transaction{t1, t2}
+
+	for _, transaction := range transactions {
+		mockDepth.EXPECT().AddTrade(transaction.Price, uint64(-transaction.StockQuantity), transaction.CreatedAt).Times(1)
+	}
+
+	ob.LoadOldTransactions(transactions)
+}
+
+func TestAddAskToDepth(t *testing.T) {
+
+	config := utils.GetConfiguration()
+	utils.Init(config)
+
+	mockControl, ob, _, _, _, _, mockDepth, stockID, stockQuantity, stockPrice := helperForOrderBookTests(t)
+	defer mockControl.Finish()
+
+	limitAsk := makeAsk(1, stockID, models.Limit, stockQuantity, stockPrice, "")
+	askUnfulfilledQuantity := limitAsk.StockQuantity - limitAsk.StockQuantityFulfilled
+
+	mockDepth.EXPECT().AddOrder(isMarket(limitAsk.OrderType), true, limitAsk.Price, askUnfulfilledQuantity)
+
+	ob.addAskToDepth(limitAsk)
+}
+
+func TestAddBidToDepth(t *testing.T) {
+
+	config := utils.GetConfiguration()
+	utils.Init(config)
+
+	mockControl, ob, _, _, _, _, mockDepth, stockID, stockQuantity, stockPrice := helperForOrderBookTests(t)
+	defer mockControl.Finish()
+
+	limitBid := makeBid(1, stockID, models.Limit, stockQuantity, stockPrice, "")
+	bidUnfulfilledQuantity := limitBid.StockQuantity - limitBid.StockQuantityFulfilled
+
+	mockDepth.EXPECT().AddOrder(isMarket(limitBid.OrderType), false, limitBid.Price, bidUnfulfilledQuantity)
+
+	ob.addBidToDepth(limitBid)
+}
+
+func TestOrderBookAddAskOrder(t *testing.T) {
```

Comment:
There is AddAskOrder function belonging to orderBook as well as matchingEngine. so i have already written a test named TestAddAskOrder to matchingEngine. Using the same name here will lead to redeclaration error since both the tests belong to matchingEngine package.